semantic/monotype.go

```diff
@@ -906,6 +913,21 @@ func NewDictType(keyType, valueType MonoType) MonoType {
 	return mt
 }
 
+// NewDynamicType will construct a new Dynamic MonoType
+func NewDynamicType() MonoType {
+	builder := flatbuffers.NewBuilder(32) // FIXME: how big?
```

Comment:
The size of this is still an open question. I'm not that familiar with flatbuffers, but it does seem to me like it should be possible to mark this as 0-sized so long as whatever fields are associated with the node size themselves.
I'm thinking in terms of the `BaseNode` fields we see like the location of the node, etc.